spec/queue_test.js

```diff
@@ -0,0 +1,110 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import Queue from '../src/queue'
+
+chai.use(chaiChange)
+
+describe('Queue', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(Queue).to.be.a('function')
+  })
+
+  describe('enqueue()', () => {
+    it('adds an element to the back of the queue.', () => {
+      const queue = new Queue()
+      expect(() => queue.enqueue('foo'))
+        .to.alter(() => queue.elements.length, { from: 0, to: 1 })
+      expect(() => queue.enqueue('bar'))
+        .to.alter(() => queue.elements.length, { from: 1, to: 2 })
+      expect(queue.elements[queue.elements.length - 1])
+        .to.eql('bar')
+
+    })
+  })
+
+  describe('dequeue()', () => {
+    it('returns and removes the front element in the queue or null if the queue is empty.', () => {
+      const queue = new Queue()
+
+      queue.enqueue('cereal')
+      queue.enqueue('almond milk')
+      queue.enqueue('bananas')
+
+      expect(() => queue.dequeue())
+        .to.alter(() => queue.elements.length, { from: 3, to: 2 })
+        expect(() => queue.dequeue())
+          .to.alter(() => queue.elements.length, { from: 2, to: 1 })
+          expect(() => queue.dequeue())
+            .to.alter(() => queue.elements.length, { from: 1, to: 0 })
+            expect(queue.dequeue())
+              .to.be.null
+      })
+  })
+
+  describe('front()', () => {
+    it('returns the front element in queue or null if the queue is empty.', () => {
+      const queue = new Queue()
+      const emptyQueue = new Queue()
+
+      queue.enqueue('cereal')
+      queue.enqueue('almond milk')
+      queue.enqueue('bananas')
+
+      expect(queue.front())
+        .to.eql('cereal')
```

Comment:
Over and Over Again Code Reviews

`eql` is for deep equality comparisons (for example, if you are comparing an array to another array). Use `equal` here.

I prefer using `to.deep.equal` for deep equality comparisons because it provides a better indication of what the assertion is doing instead of removing vowels from the word ‘eql’ and giving it different functionality.